modules/ballerina-core/src/main/java/org/ballerinalang/model/values/BFloat.java

```diff
@@ -42,12 +42,12 @@ public long longValue() {
 
     @Override
     public float floatValue() {
-        return this.value;
+        return (float) this.value;
```

Comment:
we should return double value from this method and remove the doubleValue() method entirely.